code/__HELPERS/icons.dm

```diff
@@ -780,100 +779,86 @@ world
 
 	//Try to remove/optimize this section ASAP, CPU hog.
 	//Determines if there's directionals.
-	if(!noIcon && curdir != SOUTH)
-		var/exist = FALSE
-		var/static/list/checkdirs = list(NORTH, EAST, WEST)
-		for(var/i in checkdirs) //Not using GLOB for a reason.
-			if(length(icon_states(icon(curicon, curstate, i))))
-				exist = TRUE
-				break
-		if(!exist)
+	if(render_icon && curdir != SOUTH)
+		if (
+			!length(icon_states(icon(curicon, curstate, NORTH))) \
+			&& !length(icon_states(icon(curicon, curstate, EAST))) \
+			&& !length(icon_states(icon(curicon, curstate, WEST))) \
```

Comment:
not asking you to do this but i wonder if instead of doing this every time if we can just store the dirs for each icon in a static list so this doesnt have to be rechecked every call. its only unsafe if the dirs of any of the icon_states can change during runtime which im not sure if we do that.